metricflow/specs/specs.py

```diff
@@ -522,28 +522,33 @@ class LinkableSpecSet(SerializableDataclass):
     def as_tuple(self) -> Tuple[LinkableInstanceSpec, ...]:  # noqa: D
         return tuple(itertools.chain(self.dimension_specs, self.time_dimension_specs, self.entity_specs))
 
-    @staticmethod
-    def merge(spec_sets: Sequence[LinkableSpecSet]) -> LinkableSpecSet:
-        """Merges and dedupes the linkable specs."""
-        dimension_specs: List[DimensionSpec] = []
-        time_dimension_specs: List[TimeDimensionSpec] = []
-        entity_specs: List[EntitySpec] = []
-
-        for spec_set in spec_sets:
-            for dimension_spec in spec_set.dimension_specs:
-                if dimension_spec not in dimension_specs:
-                    dimension_specs.append(dimension_spec)
-            for time_dimension_spec in spec_set.time_dimension_specs:
-                if time_dimension_spec not in time_dimension_specs:
-                    time_dimension_specs.append(time_dimension_spec)
-            for entity_spec in spec_set.entity_specs:
-                if entity_spec not in entity_specs:
-                    entity_specs.append(entity_spec)
+    @override
+    def merge(self, other: LinkableSpecSet) -> LinkableSpecSet:
+        return LinkableSpecSet(
+            dimension_specs=self.dimension_specs + other.dimension_specs,
+            time_dimension_specs=self.time_dimension_specs + other.time_dimension_specs,
+            entity_specs=self.entity_specs + other.entity_specs,
+        )
+
+    def dedupe(self) -> LinkableSpecSet:  # noqa: D
```

Comment:
Why are we separating these? Is there ever a case where we want duplicates in this output?

If not, this feels like we're adding an old bug factory back to the codebase.